daemon/top_unix.go

```diff
@@ -5,12 +5,25 @@ package daemon
 import (
 	"fmt"
 	"os/exec"
+	"regexp"
 	"strconv"
 	"strings"
 
 	"github.com/docker/engine-api/types"
 )
 
+func validatePSArgs(psArgs string) error {
+	re := regexp.MustCompile("-o\\s*(.*)=\\s*PID")
+	group := re.FindStringSubmatch(psArgs)
```

Comment:
This is not quite sufficient, as the header parsing is using `strings.Fields` and that parses based on unicode whitespace, so you can do `docker top d0b6b1e905ae ae -o uid= PID  -o command=COMMAND` with any unicode whitespace character between the `=` and `PID`. Changing the code to split strictly on ASCII spaces seems to be the best fix.